arangod/Cluster/MaintenanceFeature.cpp

```diff
@@ -163,6 +162,10 @@ void MaintenanceFeature::beginShutdown() {
 }  // MaintenanceFeature
 
 void MaintenanceFeature::stop() {
+  // There should be no new workers.
+  // Current workers could be stuck on the condition variable.
+  // Let's wake them up now.
+  _currentCounterCondition.notify_all();
```

Comment:
I think we should hold the mutex here when we call `notify_all()`:
` std::unique_lock<std::mutex> guard(_currentCounterLock);`

Reply:
right!